spyder/plugins/editor/extensions/docstring.py

```diff
@@ -0,0 +1,851 @@
+        if return_type_re:
+            self.return_type_annotated = return_type_re.group(1)
+            text_end = text.rfind(return_type_re.group(0))
+        else:
+            self.return_type_annotated = None
+            text_end = len(text)
+
+        pos_args_start = text.find('(') + 1
+        pos_args_end = text.rfind(')', pos_args_start, text_end)
+
+        self.args_text = text[pos_args_start:pos_args_end]
+
+        args_list = self.split_args_text_to_list(self.args_text)
+        if args_list is not None:
+            self.has_info = True
+            self.split_arg_to_name_type_value(args_list)
+
+    def parse_body(self, text):
+        """Parse the function body text."""
+        re_raise = re.findall(r'[ \t]raise ([a-zA-Z0-9_]*)', text)
+        if len(re_raise) > 0:
+            self.raise_list = [x.strip() for x in re_raise]
+            # remove duplicates from list while keeping it in the order
+            # in python 2.7
+            # stackoverflow.com/questions/7961363/removing-duplicates-in-lists
+            self.raise_list = list(OrderedDict.fromkeys(self.raise_list))
+
+        re_yield = re.search(r'[ \t]yield ', text)
+        if re_yield:
+            self.has_yield = True
+
+        # get return value
+        pattern_return = r'return |yield '
+        line_list = text.split('\n')
+        is_found_return = False
+        line_return_tmp = ''
+
+        for line in line_list:
+            line = line.strip()
+
+            if is_found_return is False:
+                if re.match(pattern_return, line):
+                    is_found_return = True
+
+            if is_found_return:
+                line_return_tmp += line
+                # check the integrity of line
+                try:
+                    pos_quote = self._find_quote_position(line_return_tmp)
+
+                    if line_return_tmp[-1] == '\\':
+                        line_return_tmp = line_return_tmp[:-1]
+                        continue
+
+                    self._find_bracket_position(line_return_tmp, '(', ')',
+                                                pos_quote)
+                    self._find_bracket_position(line_return_tmp, '{', '}',
+                                                pos_quote)
+                    self._find_bracket_position(line_return_tmp, '[', ']',
+                                                pos_quote)
+                except IndexError:
+                    continue
+
+                return_value = re.sub(pattern_return, '', line_return_tmp)
+                self.return_value_in_body.append(return_value)
+
+                is_found_return = False
+                line_return_tmp = ''
+
+
+class QMenuOnlyForEnter(QMenu):
+    """
+    If a input of keyboard is the "enter key", the selected action is executed.
+    If a input of keyboard is not the "enter key", the menu is closed and
+    the input is inserted to code editor.

```

Comment:
For a multi-line docstring, per PEP 257, first line should be a summary, followed by a blank line, followed by the full description.